src/test/java/org/jabref/logic/exporter/BibtexDatabaseWriterTest.java

```diff
@@ -726,6 +726,49 @@ void writeCustomKeyPattern() throws Exception {
                 stringWriter.toString());
     }
 
+    @Test
+    void AbstractCitationKeyPatternParse() throws Exception {
+        AbstractCitationKeyPattern pattern = new DatabaseCitationKeyPattern(mock(GlobalCitationKeyPattern.class));
```

Comment:
The new tests also work for the old code.

The tests are located in the wrong class. Please navigate to `AbstractCitationKeyPattern` and press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>T</kbd> to create a test class for AbstractCitationKeyPattern.

You can also do it for `DatabaseCitationKeyPattern` if this is more convenient (as AbstractCitationKeyPattern is an abstract class)